architecture/ambient/lifecycle.md

```diff
@@ -0,0 +1,136 @@
+# Ztunnel Lifecyle
+
+This document describes the lifecycle of various aspects of Ztunnel, and its relationship to workloads and the CNI.
+
+At a high level, our goal is to provide complete connectivity to a workload throughout its entire lifetime.
+Failing to do so can be an availability risk (if we deny traffic that should succeed) or a security risk (if we allow traffic that should be denied).
+
+## High level overview
+
+At a high level, the relevant components look as such:
+
+```mermaid
+flowchart TD
+    CNIP["CNI Plugin"]
+    CNI["CNI Agent"]
+    Z["Ztunnel"]
+    Pods["Pods"]
+    CNI --Programs--> Z
+    CNI --Watches--> Pods
+    Z --Runs Within--> Pods
+    CNIP --Calls--> CNI
+```
+
+The CNI Plugin is a binary installed as a [CNI plugin](https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/) on the node.
+Kubernetes is responsible for invoking this when a Pod is being started (before the containers run).
+When this occurs, the plugin will call out to the CNI Agent to program the network.
+This includes setting up networking rules both within the pod network namespace, and also on the host network.
```

Comment:
Commenting as I read - do we have some examples with what is set on the host network - and what is set on the pod network ? The first part in particular - since it's different from normal Istio where we don't set anything on the host.

Reply:
Some of that (should be/was) in https://github.com/istio/istio/tree/master/cni/README.MD

All that is created on the host are a single iptables rule and ipset for host probes.

That does raise a good question - if this is `/architecture/ambient` - should it have that impl detail in it?

Or should that impl detail go in `cni/README.MD`, next to the thing that actually does the impl?

Reply:
I linked to CNI/readme.dm